src/internal/kvparse/kvgrammar/kvgrammar.go

```diff
@@ -0,0 +1,179 @@
+package kvgrammar
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/alecthomas/participle/v2/lexer"
+)
+
+var Lexer = lexer.Rules{
+	"Root": {
+		{
+			Name:    "Whitespace",
+			Pattern: `\s+`,
+		},
+		{
+			Name:    "DoubleQuotedString",
+			Pattern: `"`,
+			Action:  lexer.Push("DoubleQuotedString"),
+		},
+		{
+			Name:    "SingleQuotedString",
+			Pattern: `'`,
+			Action:  lexer.Push("SingleQuotedString"),
+		},
+		{
+			Name:    "JSON",
+			Pattern: `{`,
+			Action:  lexer.Push("JSON"),
+		},
+		{
+			Name:    "Terminator",
+			Pattern: `;`,
+		},
+		{
+			Name:    "Equals",
+			Pattern: `=`,
+		},
+		{
+			Name:    "String",
+			Pattern: `[^;="'\s]+`,
+		},
```

Comment:
I think that allowing ` ` would still be unambiguous:

- `foo= bar baz` → `foo` = `bar baz`

Spaces are not currently meaningful in the grammar, so that change requires no grammar change.

Allowing `=` would also be unambiguous, but it would require a grammar change.